package/src/components/Attachment/Gallery.tsx

```diff
@@ -258,35 +259,62 @@ const GalleryWithContext = <
                   }-${colIndex}-item-${rowIndex}`}
                   {...additionalTouchableProps}
                 >
-                  <MemoizedGalleryImage
-                    resizeMode={resizeMode}
-                    style={[
-                      getGalleryImageBorderRadius({
-                        alignment,
-                        colIndex,
-                        groupStyles,
-                        hasThreadReplies,
-                        height,
-                        invertedDirections,
-                        messageText,
-                        numOfColumns,
-                        numOfRows,
-                        rowIndex,
-                        sizeConfig,
-                        threadList,
-                        width,
-                      }),
-                      image,
-                      {
-                        height: height - 1,
-                        width: width - 1,
-                      },
-                    ]}
-                    uri={url}
-                  />
+                  {type === 'video' ? (
+                    <VideoThumbnail
+                      style={[
+                        getGalleryImageBorderRadius({
+                          alignment,
+                          colIndex,
+                          groupStyles,
+                          hasThreadReplies,
+                          height,
+                          invertedDirections,
+                          messageText,
+                          numOfColumns,
+                          numOfRows,
+                          rowIndex,
+                          sizeConfig,
+                          threadList,
+                          width,
+                        }),
+                        image,
+                        {
+                          height: height - 1,
+                          width: width - 1,
+                        },
+                      ]}
+                    />
+                  ) : (
+                    <MemoizedGalleryImage
+                      resizeMode={resizeMode}
+                      style={[
+                        getGalleryImageBorderRadius({
+                          alignment,
+                          colIndex,
+                          groupStyles,
+                          hasThreadReplies,
+                          height,
+                          invertedDirections,
+                          messageText,
+                          numOfColumns,
+                          numOfRows,
+                          rowIndex,
+                          sizeConfig,
+                          threadList,
+                          width,
+                        }),
+                        image,
+                        {
+                          height: height - 1,
+                          width: width - 1,
+                        },
+                      ]}
+                      uri={url}
```

Comment:
I might be seeing it wrong, but I'd think we could extract the `getGalleryImageBorderRadius` call out, and avoid doing the same thing twice here?

Reply:
In that case, we will have to create a function and pass index and stuff as param to the function. Does that make sense or this looks good?